opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -275,7 +275,7 @@ def collect(
         """
         with self._lock:
             if not any(self._bucket_counts):
-                return None
+                return self._previous_point
```

Comment:
This is incorrect; we can't return the previous point always.

Reply:
Could you please suggest what cases should I consider to make the correct change?

Reply:
I will look closely at the linked issue to see the correct fix, but the issue with returning `self._previous_point` is when the temporality is delta and no measurement is recorded, it will return an incorrect result.

Reply:
I don't know if this helps, but I'm currently trying this approach, so the metrics are always present, but possibly with zero values:

```diff
diff --git a/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py b/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py
index ae21db90..25f03f69 100644
--- a/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py
+++ b/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py
@@ -289,9 +289,6 @@ class _ExplicitBucketHistogramAggregation(_Aggregation[HistogramPoint]):
         Atomically return a point for the current value of the metric.
         """
         with self._lock:
-            if not any(self._bucket_counts):
-                return None
-
             bucket_counts = self._bucket_counts
             start_time_unix_nano = self._start_time_unix_nano
             sum_ = self._sum
@@ -316,6 +313,9 @@ class _ExplicitBucketHistogramAggregation(_Aggregation[HistogramPoint]):
             max=max_,
         )

+        if not any(bucket_counts):
+            return current_point
+
         if self._previous_point is None or (
             self._instrument_temporality is aggregation_temporality
         ):
```

Reply:
Hrm, never mind, returning zero values like this does not fix my issue, that the `count` in Prometheus shows lots of wrong, way too large values.

Reply:
> I recently came back to this, and have edited my above Diff: Also resetting `_previous_point` seems to do the right thing for me.

I'm not familiar with the codebase but I'm pretty interested into getting this fixed. If someone that's able to review is still actively contributing around here, I can invest some time trying to figure it out.